WireAnalytics/Sources/WireAnalytics/Service/AnalyticsService.swift

```diff
@@ -0,0 +1,219 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Countly
+import Foundation
+
+/// A service for tracking analytics data generated by the user.
+
+public final class AnalyticsService: AnalyticsServiceProtocol {
+
+    public typealias Config = (secretKey: String, serverHost: URL)
+
+    private let config: Config?
+    private let countlyProvider: () -> any CountlyProtocol
+    private var countly: (any CountlyProtocol)?
+    private var currentUser: AnalyticsUser?
+    private let baseSegmentation: Set<SegmentationEntry>
+
+    // MARK: - Life cycle
+
+    /// Create a new `AnalyticsService`.
+    ///
+    /// A non-nil config is needed to track analytics data.
+    ///
+    /// - Parameters:
+    ///   - config: A config containing the authentication key and server host.
+
+    public convenience init(config: Config?) {
+        self.init(
+            config: config,
+            baseSegmentation: [
+                .deviceModel(UIDevice.current.model),
+                .deviceOS(UIDevice.current.systemVersion)
+            ],
+            countlyProvider: { Countly.sharedInstance() }
+        )
+    }
+
+    init(
+        config: Config?,
+        baseSegmentation: Set<SegmentationEntry>,
+        countlyProvider: @escaping () -> any CountlyProtocol
+    ) {
+        self.config = config
+        self.baseSegmentation = baseSegmentation
+        self.countlyProvider = countlyProvider
+    }
+
+    // MARK: - Enable / disable
+
+    /// Start sending analytics data.
+
+    public func enableTracking() throws {
+        guard let config else {
+            throw AnalyticsServiceError.serviceIsNotConfigured
+        }
+
+        print("[ANALYTICS] enabling tracking")
+
+        let countly = countlyProvider()
+        self.countly = countly
+
+        countly.start(
+            appKey: config.secretKey,
+            host: config.serverHost
+        )
+    }
+
+    /// Stop sending analytics data.
+
+    public func disableTracking() throws {
+        guard let countly else {
+            throw AnalyticsServiceError.serviceIsNotConfigured
+        }
+
+        print("[ANALYTICS] disabling tracking")
+
+        countly.endSession()
+        clearCurrentUser()
+        countly.resetInstance()
+        self.countly = nil
+    }
+
+    // MARK: - User
+
+    /// Switch the current analytics user.
+    ///
+    /// - Parameter user: The user to switch to.
+
+    public func switchUser(_ user: AnalyticsUser) {
+        guard
+            let countly,
+            user != currentUser
+        else {
+            return
+        }
+
+        print("[ANALYTICS] switching user")
```

Comment:
We don't have `WireLogger` available here, we also can't depend on it yet since it's part of an Xcode project. Perhaps  we can move it to a new target in `WireFoundation`. Thoughts?

Reply:
there's a PR https://github.com/wireapp/wire-ios/pull/1938, it's not for 3.113 branch though.
temp solution could be inject a function in AnalyticsService

Reply:
or port WireFoundation to 3.113 but sounds like a big thing to port

Reply:
Ok I'll inject a log function for now then switch to WireLogger later.